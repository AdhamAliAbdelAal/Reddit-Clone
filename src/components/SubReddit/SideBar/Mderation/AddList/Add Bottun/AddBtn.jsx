import ClearOutlinedIcon from '@mui/icons-material/ClearOutlined';
import ModeEditOutlineOutlinedIcon from '@mui/icons-material/ModeEditOutlineOutlined';
import {
  Action,
  Add, BOX, Count, Input, InputFooter, Text,
} from './style';

const { ClickAwayListener, Box } = require('@mui/material');

function AddBtn(props) {
  const { sendData } = props;
  const {
    handleClickAway1, falseShaw, show, tempString, removeItem, tags, handleChange, handleKeyDown, trueShawList, trueShaw,
    setTemp, setTag, count,
  } = props;
  return (
    <ClickAwayListener onClickAway={handleClickAway1}>
      <Add data-testid="add" condition={show?.toString()} onClick={() => { falseShaw(); }} id="add">
        <>
          <Box sx={{ display: 'flex', flexDirection: 'row', flexWrap: 'wrap' }}>
            {tempString?.length > 0 && (tempString.slice(0, 4))?.map((tag, index) => (
              <Text key={`${index + 0}`} condition={show?.toString()}>
                {tag}
                <ClearOutlinedIcon
                  fontSize="small"
                  onClick={() => {
                    removeItem(index);
                  }}
                />
              </Text>
            )) }
            { !show && tags?.length > 0
              && (tempString.slice(4))?.map((temp, index) => (
                <Text key={`${index + 0}`} condition={show?.toString()}>
                  {temp}
                  <ClearOutlinedIcon
                    fontSize="small"
                    onClick={() => {
                      removeItem(index + 4);
                    }}
                    sx={{ color: 'gray' }}
                  />
                </Text>
              )) }
            {show && tempString?.length > 4 && (
            <BOX>
              +
              {tempString.length - 4}
            </BOX>
            )}
            {show && <ModeEditOutlineOutlinedIcon />}
          </Box>
          {!show
            && (
            <Input
              data-testid="input"
              autoFocus
              id="input"
              type="text"
              onChange={handleChange}
              onFocus={trueShawList}
              onInput={(e) => {
                // eslint-disable-next-line radix
                e.target.style.height = `${e.target.scrollHeight}px`;
              }}
              onKeyDown={handleKeyDown}
            />
            )}
        </>
        {!show && (
        <InputFooter>
          <Count
            condition={(count === 25).toString()}
          >
            {tempString?.length}
            /
            25
          </Count>
          <Box sx={{ display: 'flex', marginTop: '1px' }}>
            <Action
              onClick={(e) => {
                e.stopPropagation();
                trueShaw();
                setTemp(tags);
                setTemp(tags);
                sendData();
              }}
              id="cancel"
              color="red"
              sx={{ marginRight: '8px' }}
            >
              Cancel
            </Action>
            <Action
              color="#0079d3"
              onClick={(e) => {
                e.stopPropagation();
                trueShaw();
                setTag(tempString);
                setTemp(tempString);
                sendData();
              }}
            >
              Save
            </Action>
          </Box>
        </InputFooter>
        )}
      </Add>
    </ClickAwayListener>
  );
}
export default AddBtn;
