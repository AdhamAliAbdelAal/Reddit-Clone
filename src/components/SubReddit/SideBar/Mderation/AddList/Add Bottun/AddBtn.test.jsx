import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import AddBtn from './AddBtn';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<AddBtn />).toJSON();
  expect(tree).toMatchSnapshot();
});
