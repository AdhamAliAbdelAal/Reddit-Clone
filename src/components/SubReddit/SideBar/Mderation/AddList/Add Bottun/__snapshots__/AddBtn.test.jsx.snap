// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<span
  className="css-19lxpi3"
  data-testid="add"
  id="add"
  onClick={[Function]}
  onTouchEnd={[Function]}
>
  <div
    className="MuiBox-root css-p58oka"
  />
  <textarea
    autoFocus={true}
    className="css-h1mzer"
    data-testid="input"
    id="input"
    onInput={[Function]}
    type="text"
  />
  <div
    className="MuiBox-root css-1e5p0dr"
  >
    <p
      className="css-13icyxo"
    >
      / 25
    </p>
    <div
      className="MuiBox-root css-1euf48e"
    >
      <p
        className="css-1w2jjrd"
        color="red"
        id="cancel"
        onClick={[Function]}
      >
        Cancel
      </p>
      <p
        className="css-44rl5i"
        color="#0079d3"
        onClick={[Function]}
      >
        Save
      </p>
    </div>
  </div>
</span>
`;
